src/wrapper/Websocket.ts

```diff
@@ -185,10 +185,17 @@ export default class Websocket {
                 const baseUrl = (
                     (await core.Supplier.get(this.options.environment)) ?? environments.CartesiaEnvironment.Production
                 ).replace(/^https?:\/\//, "");
-                const params = {
-                    api_key: await core.Supplier.get(this.options.apiKey),
-                    cartesia_version: this.options.cartesiaVersion,
+                const params: Record<string, string> = {
+                    cartesia_version: this.options.cartesiaVersion || "2024-06-10",
                 };
+                if (options.accessToken) {
+                    params.access_token = options.accessToken;
+                } else {
+                    const apiKey = await core.Supplier.get(this.options.apiKey);
```

Comment:
shouldn't `apiKey` have precedence over `accessToken` since an `apiKey` can be used to generate tokens?

Reply:
Perhaps. I saw the `await` and decided to optimize for the non-`await` path.

Reply:
It's only awaited because of the auto-generated tangle (as you mentioned)

Reply:
ok I rewrote it to try apiKey first